driver/src/contracts/stablex_contract.rs

```diff
@@ -160,15 +202,14 @@ impl StableXContract for StableXContractImpl {
         page_size: u16,
         previous_page_user: Address,
         previous_page_user_offset: u16,
+        block_number: Option<BlockNumber>,
```

Comment:
maybe the change to this method should be reverted since we shouldn't use it anymore?

Reply:
I feel the Reader should be in charge of setting the block (like in the method above). How about removing the .or(Some(BlockNumber::Pending)) from here and moving it to the PaginatedOrderbookReader so that the two APIs are consistent?